src/formatDate.js

```diff
@@ -50,7 +50,78 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldFormatDelimeter = fromFormat[3];
+  const newFormatDelimeter = toFormat[3];
+  const newDate = date.split(oldFormatDelimeter);
+  const newDateArray = [];
+  
+  let oldYearLength;
+  let oldYearPosition;
+  let oldMonthPostion;
```

Comment:
you have no need to write this variables, in fact, you need only oldYear and newYear